src/FuelClient.cc

```diff
@@ -1566,3 +1585,85 @@ void FuelClientPrivate::PopulateLicenses(const ServerConfig &_server)
     ignerr << "Failed to parse license information[" << resp.data << "]\n";
   }
 }
+
+//////////////////////////////////////////////////
+bool FuelClient::UpdateModels(const std::vector<std::string> &_headers)
+{
+  // Get a list of the most recent model versions in the cache.
+  std::map<std::string, ignition::fuel_tools::ModelIdentifier> toProcess;
+  for (ModelIter iter = this->dataPtr->cache->AllModels(); iter; ++iter)
+  {
+    auto processIter = toProcess.find(iter->Identification().UniqueName());
+    if (processIter == toProcess.end() ||
+        processIter->second.Version() < iter->Identification().Version())
+    {
+      toProcess[iter->Identification().UniqueName()] = iter->Identification();
+    }
+  }
+
+  // Attempt to update each model.
+  for (const auto id : toProcess)
+  {
+    ignition::fuel_tools::ModelIdentifier cloudId;
+
+    if (!this->ModelDetails(id.second, cloudId, _headers))
+    {
+      ignerr << "Failed to fetch model details for model["
+        << id.second.Owner()  << "/" << id.second.Name() << "]\n";
+    }
+    else if (id.second.Version() < cloudId.Version())
+    {
+      ignmsg << "Updating model " << id.second.Owner() << "/"
+        << id.second.Name() << " up to version "
+        << cloudId.Version() << std::endl;
+      this->DownloadModel(cloudId, _headers);
+    }
+    else
+    {
+      ignmsg << "Model " << id.second.Owner() << "/"
+        << id.second.Name() << " is up to date." << std::endl;
+    }
+  }
+  return true;
```

Comment:
Is it worth to keep it a boolean function if it can only return true? Maybe moving it to a void function?

Reply:
As documentation states ```/// \return True if everything updated successfully.```, but it always returns true

Reply:
I'm fine keeping the return statement since it future proofs this function with minimal maintenance overhead.